scripts/pylib/pytest-twister-harness/src/twister_harness/device/device_adapter.py

```diff
@@ -0,0 +1,264 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import abc
+import logging
+import os
+import queue
+import shutil
+import threading
+import time
+from datetime import datetime
+from pathlib import Path
+
+from twister_harness.constants import END_OF_DATA
+from twister_harness.exceptions import (
+    TwisterHarnessException,
+    TwisterHarnessTimeoutException,
+)
+from twister_harness.twister_harness_config import DeviceConfig
+
+logger = logging.getLogger(__name__)
+
+
+class DeviceAdapter(abc.ABC):
+    """Class defines an interface for all devices."""
+
+    def __init__(self, device_config: DeviceConfig, **kwargs) -> None:
+        """
+        :param device_config: device configuration
+        """
+        self.device_config: DeviceConfig = device_config
+        self.base_timeout: float = device_config.base_timeout
+        self._device_read_queue: queue.Queue = queue.Queue()
+        self._reader_thread: threading.Thread | None = None
+        self._device_run: threading.Event = threading.Event()
+        self._device_connected: threading.Event = threading.Event()
+        self._end_of_data_detected: bool = True
+        self.command: list[str] = []
+
+        self.handler_log_path: Path = Path(device_config.build_dir) / 'handler.log'
+        self.device_log_path: Path = Path(device_config.build_dir) / 'device.log'
```

Comment:
device.log był dodawany tylko dla hardware, może nie ma sensu trzymać tego tutaj.

Reply:
dobre pytanie. generalnie build/device/handler logi są cechą workflow twistera i tam nie ma rozróżnienia, co powinno mieć a co nie danego loga. chyba bym zostawił. może np do renoda się przyda? jak będzie jakiś problem z wywołaniem renoda a nie samego testu. Myślałem, czy z qemu nie mogłoby być podobnie, że jak jest z qemu problem to tam trafia, ale tego chyba nie potrafimy rozróżnić. west -t run jak się wywali to raczej problem aplikacji niż qemu